src/erlcloud_config.erl

```diff
@@ -0,0 +1,632 @@
+-define(CHRONOLOGICAL_ORDER, <<"chronologicalOrder">>).
+-define(EARLIER_TIME, <<"earlierTime">>).
+-define(LATER_TIME, <<"laterTime">>).
+-define(INCLUDE_DELETED_RESOURCES, <<"includeDeletedResources">>).
+
+
+%%------------------------------------------------------------------------------
+%% Library initialization.
+%%------------------------------------------------------------------------------
+
+-spec new(string(), string()) -> aws_config().
+new(AccessKeyID, SecretAccessKey) ->
+    #aws_config{access_key_id     = AccessKeyID,
+                secret_access_key = SecretAccessKey,
+                retry             = fun erlcloud_retry:default_retry/1}.
+
+-spec new(string(), string(), string()) -> aws_config().
+new(AccessKeyID, SecretAccessKey, Host) ->
+    #aws_config{access_key_id     = AccessKeyID,
+                secret_access_key = SecretAccessKey,
+                config_host       = Host,
+                retry             = fun erlcloud_retry:default_retry/1}.
+
+-spec new(string(), string(), string(), non_neg_integer()) -> aws_config().
+new(AccessKeyID, SecretAccessKey, Host, Port) ->
+    #aws_config{access_key_id     = AccessKeyID,
+                secret_access_key = SecretAccessKey,
+                config_host       = Host,
+                config_port       = Port,
+                retry             = fun erlcloud_retry:default_retry/1}.
+
+-spec new(string(), string(), string(), non_neg_integer(), string()) ->
+    aws_config().
+new(AccessKeyID, SecretAccessKey, Host, Port, Scheme) ->
+    #aws_config{access_key_id     = AccessKeyID,
+                secret_access_key = SecretAccessKey,
+                config_host       = Host,
+                config_port       = Port,
+                config_scheme     = Scheme,
+                retry             = fun erlcloud_retry:default_retry/1}.
+
+-spec configure(string(), string()) -> ok.
+configure(AccessKeyID, SecretAccessKey) ->
+    put(aws_config, new(AccessKeyID, SecretAccessKey)),
+    ok.
+
+-spec configure(string(), string(), string()) -> ok.
+configure(AccessKeyID, SecretAccessKey, Host) ->
+    put(aws_config, new(AccessKeyID, SecretAccessKey, Host)),
+    ok.
+
+-spec configure(string(), string(), string(), non_neg_integer()) -> ok.
+configure(AccessKeyID, SecretAccessKey, Host, Port) ->
+    put(aws_config, new(AccessKeyID, SecretAccessKey, Host, Port)),
+    ok.
+
+-spec configure(string(), string(), string(), non_neg_integer(), string()) -> ok.
+configure(AccessKeyID, SecretAccessKey, Host, Port, Scheme) ->
+    put(aws_config, new(AccessKeyID, SecretAccessKey, Host, Port, Scheme)),
+    ok.
+
+%%------------------------------------------------------------------------------
+%% AWS Config API Functions
+%%------------------------------------------------------------------------------
+
+%%------------------------------------------------------------------------------
+%% @doc
+%% DescribeComplianceByConfigRule
+%%
```

Comment:
@ikemsley I think what makes the description really good, is an example of input (especially if it's map with optional arguments). Could you please add it there?

Example: https://github.com/erlcloud/erlcloud/blob/b50123a62da79/src/erlcloud_glue.erl#L184-L186

Reply:
Or actually, it could be seen in the tests module. probably not the best idea to duplicate it everywhere then. 

So please disregard my previous comment :)